### ISSUE TITLE
CLI - new options added to hammer

### ISSUE THREAD

**Issue by rplevka** [SatelliteQE/robottelo#5224]
The following options have been added to hammer. We might need to extend our test suite:

```
    @tier1
    @upgrade
    def test_positive_all_options(self):
        """check all provided options for every hammer command
    
            @id: 1203ab9f-896d-4039-a166-9e2d36925b5b
    
            @expectedresults: All expected options are present
            """
        self.maxDiff = None
        self._traverse_command_tree('hammer')
        if self.differences:
            self.fail(
>               '\n' + _format_commands_diff(self.differences)
            )
E           AssertionError: 
E           hammer capsule content synchronize
E             Added options:
E               * skip-metadata-check
E           
E           hammer host list
E             Added options:
E               * thin
E           
E           hammer host-collection hosts
E             Added options:
E               * thin
E           
E           hammer proxy content synchronize
E             Added options:
E               * skip-metadata-check
```

**Comment by rplevka** [comment]
update:
```
E           AssertionError: 
E           hammer arf-report delete
E             Removed options:
E               * name
E           
E           hammer auth-source ldap create
E             Added options:
E               * use-netgroups
E           
E           hammer auth-source ldap update
E             Added options:
E               * use-netgroups
E           
E           hammer host list
E             Added options:
E               * thin
E           
E           hammer host-collection hosts
E             Added options:
E               * thin
E           
E           hammer scap-content
E             Added subcommands:
E               * download
E           
E           hammer scap-content download (new command)
E             Added options:
E               * path
E               * title
E               * help
E               * id
```

**Comment by rplevka** [comment]
E           AssertionError: 
E           hammer arf-report delete
E             Removed options:
E               * name
E           
E           hammer auth-source ldap create
E             Added options:
E               * organization-titles
E               * use-netgroups
E               * location-titles
E           
E           hammer auth-source ldap list
E             Added options:
E               * organization-title
E               * location-title
E           
E           hammer auth-source ldap update
E             Added options:
E               * organization-titles
E               * use-netgroups
E               * location-titles
E           
E           hammer capsule create
E             Added options:
E               * organization-titles
E               * location-titles
E           
E           hammer capsule list
E             Added options:
E               * organization-title
E               * location-title
E           
E           hammer capsule update
E             Added options:
E               * organization-titles
E               * location-titles
E           
E           hammer compute-resource
E             Added subcommands:
E               * networks
E           
E           hammer compute-resource create
E             Added options:
E               * organization-titles
E               * location-titles
E           
E           hammer compute-resource list
E             Added options:
E               * organization-title
E               * location-title
E           
E           hammer compute-resource networks (new command)
E             Added options:
E               * cluster-id
E               * help
E               * name
E               * id
E           
E           hammer compute-resource update
E             Added options:
E               * organization-titles
E               * location-titles
E           
E           hammer content-view filter create
E             Added options:
E               * product
E               * product-id
E           
E           hammer discovery-rule create
E             Added options:
E               * organization-titles
E               * location-titles
E           
E           hammer discovery-rule update
E             Added options:
E               * organization-titles
E               * location-titles
E           
E           hammer docker container create
E             Added options:
E               * organization-titles
E               * location-titles
E           
E           hammer docker registry create
E             Added options:
E               * organization-titles
E               * location-titles
E           
E           hammer docker registry update
E             Added options:
E               * organization-titles
E               * location-titles
E           
E           hammer domain create
E             Added options:
E               * organization-titles
E               * location-titles
E           
E           hammer domain list
E             Added options:
E               * organization-title
E               * location-title
E           
E           hammer domain update
E             Added options:
E               * organization-titles
E               * location-titles
E           
E           hammer environment create
E             Added options:
E               * organization-titles
E               * location-titles
E           
E           hammer environment list
E             Added options:
E               * organization-title
E               * location-title
E           
E           hammer environment update
E             Added options:
E               * organization-titles
E               * location-titles
E           
E           hammer filter create
E             Added options:
E               * organization-titles
E               * location-titles
E           
E           hammer filter update
E             Added options:
E               * organization-titles
E               * location-titles
E           
E           hammer full-help (new command)
E             Added options:
E               * md
E               * help
E           
E           hammer host create
E             Added options:
E               * organization-title
E               * location-title
E           
E           hammer host list
E             Added options:
E               * organization-title
E               * thin
E               * location-title
E           
E           hammer host update
E             Added options:
E               * organization-title
E               * location-title
E           
E           hammer host-collection hosts
E             Added options:
E               * hostgroup-title
E               * location-title
E               * thin
E           
E           hammer hostgroup create
E             Added options:
E               * organization-titles
E               * location-titles
E           
E           hammer hostgroup list
E             Added options:
E               * organization-title
E               * location-title
E           
E           hammer hostgroup update
E             Added options:
E               * organization-titles
E               * location-titles
E           
E           hammer job-template create
E             Added options:
E               * organization-titles
E               * location-titles
E           
E           hammer job-template list
E             Added options:
E               * organization-title
E               * location-title
E           
E           hammer job-template update
E             Added options:
E               * organization-titles
E               * location-titles
E           
E           hammer location add-compute-resource
E             Added options:
E               * title
E           
E           hammer location add-config-template
E             Added options:
E               * title
E           
E           hammer location add-domain
E             Added options:
E               * title
E           
E           hammer location add-environment
E             Added options:
E               * title
E           
E           hammer location add-hostgroup
E             Added options:
E               * title
E           
E           hammer location add-medium
E             Added options:
E               * title
E           
E           hammer location add-organization
E             Added options:
E               * organization-title
E               * title
E           
E           hammer location add-smart-proxy
E             Added options:
E               * title
E           
E           hammer location add-subnet
E             Added options:
E               * title
E           
E           hammer location add-user
E             Added options:
E               * title
E           
E           hammer location delete
E             Added options:
E               * title
E           
E           hammer location delete-parameter
E             Added options:
E               * location-title
E           
E           hammer location info
E             Added options:
E               * title
E           
E           hammer location remove-compute-resource
E             Added options:
E               * title
E           
E           hammer location remove-config-template
E             Added options:
E               * title
E           
E           hammer location remove-domain
E             Added options:
E               * title
E           
E           hammer location remove-environment
E             Added options:
E               * title
E           
E           hammer location remove-hostgroup
E             Added options:
E               * title
E           
E           hammer location remove-medium
E             Added options:
E               * title
E           
E           hammer location remove-organization
E             Added options:
E               * organization-title
E               * title
E           
E           hammer location remove-smart-proxy
E             Added options:
E               * title
E           
E           hammer location remove-subnet
E             Added options:
E               * title
E           
E           hammer location remove-user
E             Added options:
E               * title
E           
E           hammer location set-parameter
E             Added options:
E               * location-title
E           
E           hammer location update
E             Added options:
E               * title
E           
E           hammer medium create
E             Added options:
E               * organization-titles
E               * location-titles
E           
E           hammer medium list
E             Added options:
E               * organization-title
E               * location-title
E           
E           hammer medium update
E             Added options:
E               * organization-titles
E               * location-titles
E           
E           hammer organization add-compute-resource
E             Added options:
E               * title
E           
E           hammer organization add-config-template
E             Added options:
E               * title
E           
E           hammer organization add-domain
E             Added options:
E               * title
E           
E           hammer organization add-environment
E             Added options:
E               * title
E           
E           hammer organization add-hostgroup
E             Added options:
E               * title
E           
E           hammer organization add-location
E             Added options:
E               * location-title
E               * title
E           
E           hammer organization add-medium
E             Added options:
E               * title
E           
E           hammer organization add-smart-proxy
E             Added options:
E               * title
E           
E           hammer organization add-subnet
E             Added options:
E               * title
E           
E           hammer organization add-user
E             Added options:
E               * title
E           
E           hammer organization create
E             Added options:
E               * hostgroup-titles
E           
E           hammer organization delete
E             Added options:
E               * title
E           
E           hammer organization delete-parameter
E             Added options:
E               * organization-title
E           
E           hammer organization info
E             Added options:
E               * title
E           
E           hammer organization remove-compute-resource
E             Added options:
E               * title
E           
E           hammer organization remove-config-template
E             Added options:
E               * title
E           
E           hammer organization remove-domain
E             Added options:
E               * title
E           
E           hammer organization remove-environment
E             Added options:
E               * title
E           
E           hammer organization remove-hostgroup
E             Added options:
E               * title
E           
E           hammer organization remove-location
E             Added options:
E               * location-title
E               * title
E           
E           hammer organization remove-medium
E             Added options:
E               * title
E           
E           hammer organization remove-smart-proxy
E             Added options:
E               * title
E           
E           hammer organization remove-subnet
E             Added options:
E               * title
E           
E           hammer organization remove-user
E             Added options:
E               * title
E           
E           hammer organization set-parameter
E             Added options:
E               * organization-title
E           
E           hammer organization update
E             Added options:
E               * title
E           
E           hammer package list
E             Added options:
E               * content-view-id
E               * content-view
E           
E           hammer partition-table create
E             Added options:
E               * organization-titles
E               * location-titles
E           
E           hammer partition-table list
E             Added options:
E               * organization-title
E               * location-title
E           
E           hammer partition-table update
E             Added options:
E               * organization-titles
E               * location-titles
E           
E           hammer proxy create
E             Added options:
E               * organization-titles
E               * location-titles
E           
E           hammer proxy list
E             Added options:
E               * organization-title
E               * location-title
E           
E           hammer proxy update
E             Added options:
E               * organization-titles
E               * location-titles
E           
E           hammer realm create
E             Added options:
E               * organization-titles
E               * location-titles
E           
E           hammer realm list
E             Added options:
E               * organization-title
E               * location-title
E           
E           hammer realm update
E             Added options:
E               * organization-titles
E               * location-titles
E           
E           hammer role clone
E             Added options:
E               * organization-titles
E               * location-titles
E           
E           hammer role create
E             Added options:
E               * organization-titles
E               * location-titles
E           
E           hammer role update
E             Added options:
E               * organization-titles
E               * location-titles
E           
E           hammer scap-content
E             Added subcommands:
E               * download
E           
E           hammer scap-content download (new command)
E             Added options:
E               * path
E               * title
E               * help
E               * id
E           
E           hammer subnet
E             Added subcommands:
E               * delete-parameter
E               * set-parameter
E           
E           hammer subnet create
E             Added options:
E               * organization-titles
E               * location-titles
E             Removed options:
E               * subnet-parameters-attributes
E           
E           hammer subnet delete-parameter (new command)
E             Added options:
E               * subnet
E               * name
E               * subnet-id
E               * help
E           
E           hammer subnet list
E             Added options:
E               * organization-title
E               * location-title
E           
E           hammer subnet set-parameter (new command)
E             Added options:
E               * subnet
E               * name
E               * subnet-id
E               * hidden-value
E               * value
E               * help
E           
E           hammer subnet update
E             Added options:
E               * organization-titles
E               * location-titles
E             Removed options:
E               * subnet-parameters-attributes
E           
E           hammer template create
E             Added options:
E               * organization-titles
E               * location-titles
E           
E           hammer template list
E             Added options:
E               * organization-title
E               * location-title
E           
E           hammer template update
E             Added options:
E               * organization-titles
E               * location-titles
E           
E           hammer user create
E             Added options:
E               * organization-titles
E               * ask-password
E               * location-titles
E           
E           hammer user list
E             Added options:
E               * organization-title
E               * location-title
E           
E           hammer user update
E             Added options:
E               * organization-titles
E               * ask-password
E               * location-titles
E           
E           hammer virt-who-config create
E             Added options:
E               * organization-title
E           
E           hammer virt-who-config update
E             Added options:
E               * organization-title

**Comment by san7ket** [comment]
Bump , can this be taken up by someone

**Comment by rplevka** [comment]
update.
some bootdisk-related options have been added too:

```
E           AssertionError: 
E           hammer bootdisk
E             Added subcommands:
E               * subnet
E           
E           hammer bootdisk subnet (new command)
E             Added options:
E               * subnet
E               * force
E               * help
E               * sudo
E               * subnet-id
E               * file
E           
E           hammer file (new command)
E             Added subcommands:
E               * info
E               * list
E             Added options:
E               * help
E           
E           hammer file info (new command)
E             Added options:
E               * content-view-id
E               * product
E               * product-id
E               * name
E               * repository
E               * content-view
E               * organization-label
E               * content-view-version
E               * organization-id
E               * content-view-version-id
E               * organization
E               * repository-id
E               * id
E               * help
E           
E           hammer file list (new command)
E             Added options:
E               * product-id
E               * help
E               * content-view-version
E               * by
E               * environment
E               * repository-id
E               * product
E               * repository
E               * content-view-filter-id
E               * content-view-id
E               * organization-id
E               * content-view-version-id
E               * environment-id
E               * content-view
E               * search
E               * organization-label
E               * content-view-filter
E               * ids
E               * order
E               * full-result
E               * organization
E               * page
E               * per-page
E           
E           hammer host create
E             Added options:
E               * openscap-proxy-id
E           
E           hammer host list
E             Removed options:
E               * include
E           
E           hammer host update
E             Added options:
E               * openscap-proxy-id
E           
E           hammer host-collection hosts
E             Added options:
E               * thin
E           
E           hammer job-invocation info
E             Removed options:
E               * name
E           
E           hammer job-template
E             Added subcommands:
E               * import
E               * export
E           
E           hammer job-template export (new command)
E             Added options:
E               * help
E               * name
E               * id
E           
E           hammer job-template import (new command)
E             Added options:
E               * file
E               * overwrite
E               * help
E           
E           hammer repository-set disable
E             Removed options:
E               * new-name
E           
E           hammer repository-set enable
E             Removed options:
E               * new-name
```

**Comment by ntkathole** [comment]
Closed via PR #5748
